### ISSUE TITLE
Better instructions for installing travis command line client

### ISSUE THREAD

**Issue by glyphobet** [travis-ci/travis-ci#2055]
TL;DR; Add `curl -sSL https://get.rvm.io | bash -s stable --ruby` to the docs.

As a travis-ci user, I want to install the travis command line client so that I can clear the cache whenever I want. 

However, it has been years since I've used ruby for anything, and have only a vague memory of rbenv or rvm and having to sudo install a bunch of stuff to get a current version of ruby on Mac OS X. 

The installation instructions, `gem install travis`, cause a ruby process to use 99% CPU for quite a while (10+ minutes) and then gives this:

```
gem install travis
ERROR:  While executing gem ... (Gem::FilePermissionError)
    You don't have write permissions into the /Library/Ruby/Gems/1.8 directory
```

Installing ruby1.9 with MacPorts with

`sudo port install ruby19`

gives me a newer version of ruby, but then attempting to use the version of gem still tries to write to system locations:

```
/opt/local/bin/gem1.9 install travis
Fetching: multipart-post-1.2.0.gem (100%)
ERROR:  While executing gem ... (Gem::FilePermissionError)
    You don't have write permissions into the /opt/local/lib/ruby1.9/gems/1.9.1 directory.
```

Obviously I need the equivalent of Python's virtualenv in Ruby, but not being a ruby person I have no idea what this is. 

Next I tried this (suggested by Zebra Hiro A. who was very gracious to help me on the support chat for hours)

```
GEM_HOME=rvm /opt/local/bin/gem1.9 install travis
```

That failed the first time I ran it with 

```
Installing ri documentation for pry-0.9.12.6...

RDoc::Parser::Ruby failure around line 25 of
lib/pry/commands/ls.rb

Before reporting this, could you check that the file you're documenting
has proper syntax:

  /opt/local/bin/ruby1.9 -c lib/pry/commands/ls.rb

RDoc is not a full Ruby parser and will fail when fed invalid ruby programs.

The internal error was:

    (NoMethodError) undefined method `name' for #<RDoc::RubyToken::TkDOT:0x007fd979293ab0>

ERROR:  While generating documentation for pry-0.9.12.6
... MESSAGE:   undefined method `name' for #<RDoc::RubyToken::TkDOT:0x007fd979293ab0>
... RDOC args: --ri --op rvm/doc/pry-0.9.12.6/ri lib --title pry-0.9.12.6 Documentation --quiet
```

And it succeeded the second time I ran it.

Next I tried to run the travis client, and got this: https://gist.github.com/glyphobet/85f929bf9b73ee15949e

After reviewing the shell script from get.rvm.io, I installed `rvm` with this (again suggested by Zebra Hiro A):

```
curl -sSL https://get.rvm.io | bash -s stable --ruby
source /Users/matt/.rvm/scripts/rvm
gem install travis
```

After all this, it seems all that was missing from the instructions was the first two lines of this, but it took two hours to get to that.

**Comment by OJFord** [comment]
@roidrage 
I want to just do:
```sh
brew install travis
```

I can do that without checking docs. I was `brew search`ing 'travis', 'travisci', 'traviscli'; wondering why I couldn't find it before I came here.

**Comment by BanzaiMan** [comment]
@OJFord We do not provide a Homebrew package at this time.

**Comment by OJFord** [comment]
@BanzaiMan Yes, I know.

My point is that here you have an issue with someone struggling to install the tool; it could be as easy as `brew install traviscli`.

Many people - including myself - will try `brew` prior to looking up how the author suggests installing it.

Personally I'd far rather use even a third-party tap than `gem install` it.

What's the reasoning behind not having one, when your installation instructions even point users to `brew` to install dependencies? If the package was installed via brew, we wouldn't have to worry about dependencies...

**Comment by johnwchadwick** [comment]
It looks like RVM is not quite working on macOS Seirra right now: there's no binary Ruby, and when it tries to compile it fails. In other words, there is no obvious way to get the Travis CLI working right now on Mac OS X.

**Comment by glyph** [comment]
@johnwchadwick - I am not sure what you're referring to; I just built rvm and ruby and installed travis on Sierra.  Do you have a link to an rvm bug report?

**Comment by johnwchadwick** [comment]
Some trouble running `brew update`, somewhat ironically. It runs fine for me and I don't feel like debugging, so I'll just assume it's limited to my machine somehow.

Regardless, I solved my problem by installing Ruby from Homebrew and using that instead. I feel that's less obtrusive for users who do not develop Ruby, since on macOS it's pretty unlikely you don't already use either Homebrew or at minimum one of the other package managers. So for me, the best way to install Travis's CLI is:
```
brew install ruby
/usr/local/bin/gem install travis
```

But this is not really ideal. Yes, I am a developer and likely have developer tools like Ruby installed, but, I'm also an end-user of tools like Travis CLI, Vagrant, and probably many other tools written in Ruby - and as an end user, I don't care about the programming language it's written in, and would just prefer to manage it like literally anything else I install, instead of having to bother with the details of a language-specific package manager.

**Comment by probonopd** [comment]
It is still a major hassle to get `travis` to run. Too many steps involved, files cluttering `$HOME`.

Travis CI should just provide three downloads: Windows self-contained zip, macOS self-contained `.dmg`, Linux self-contained [AppImage](http://appimage.org/).

Fully standalone bundles, including ruby, all the libraries, and the `travis` CLI.

**Comment by johnwchadwick** [comment]
I would not mind working on a homebrew formula if that's going to be a thing. But if so there probably should be a separate issue about it. Same for AppImage or a DMG.

**Comment by probonopd** [comment]
I have built an AppImage for 2014-ish and later Linux distirbutions:
https://bintray.com/probono/AppImages/travis/_latestVersion#files

This bundles ruby, rvm, gems, and `travis`.

Just download the AppImage file, [set the executable permission](http://discourse.appimage.org/t/how-to-make-an-appimage-executable/80), and run.

Let me know how it goes.

**Comment by trevordmiller** [comment]
Would also love to see a homebrew package for the CLI. Travis is now the only tool that I need Ruby on my machine for anymore.

**Comment by sanderson** [comment]
*I promise this isn't meant to be self-promotion.*

I literally just published an article that would help with this issue. I've been working on a project called [Nanobox](https://nanobox.io), a micro-platform that allows you to build and run your app anywhere. 

When using Nanobox for local development, it creates an isolated virtual environment with its own binaries and runtimes. This means no version conflicts and no cluttering up your local machine's `$HOME`. Everything gets installed inside the VM. You don't even need ruby on your host machine.

Any way, here's the post about installing the Travis CLI when using Nanobox: https://content.nanobox.io/installing-using-the-travis-cli-with-nanobox/

**Comment by Halt001** [comment]
Not being a Ruby dev and not knowing anything about it this Travis CLI install feels like something I don't want to do. Obviously it would be preferred to use Homebrew and maybe even NPM.

That said, the following solved the error on Mac OS/X 10.12.6:
```
ERROR:  While executing gem ... (Gem::FilePermissionError)
    You don't have write permissions into the /Library/Ruby/Gems/1.8 directory
```

`brew install ruby`
`gem update --system`

**Then restart the terminal.**

After that install Travis CLI as documented:
`gem install travis -v 1.8.8 --no-rdoc --no-ri`

**Comment by sharma02gaurav** [comment]
@Halt001 The solution works like charm. Should close this issue.

**Comment by rasmus-storjohann-PG** [comment]
This worked nicely for me just now:
```
sudo apt install ruby ruby-dev
sudo gem install travis
```
Found it here https://gist.github.com/jarun/df5d864f0b2600564c9bc13cfaca7497

**Comment by rob137** [comment]
Tried the instructions linked [here](https://gist.github.com/jarun/df5d864f0b2600564c9bc13cfaca7497) but still getting the following errors: 
https://gist.github.com/rob137/1dd2452f61680b245f639166367494e8

**Comment by Rooster212** [comment]
FYI - getting this to work on the Ubuntu subsystem for Windows was relatively easy with these instructions.

I did have to import the key manually but on a fresh install of Ubuntu subsystem this is working first time for me, using the following commands:
```bash
command curl -sSL https://rvm.io/mpapis.asc | gpg --import -
curl -sSL https://get.rvm.io | bash -s stable --ruby
source /home/jamie/.rvm/scripts/rvm
gem install travis
```

**Comment by jtumano** [comment]
@Rooster212 this also worked for me on Debian 9

**Comment by the-vampiire** [comment]
## OSX high sierra 10.13.4 install error and solution

### command: 
```gem install travis -v 1.8.8 --no-rdoc --no-ri```

### error:
```
ERROR:  While executing gem ... (Gem::FilePermissionError)
    You don't have write permissions for the /Library/Ruby/Gems/2.3.0 directory.
```

### solution: 

1. `sudo chown -R $USER /Library/Ruby/Gems/`
2. `gem update --system`
3. `gem install travis -v 1.8.8 --no-rdoc --no-ri`

**Comment by deadcoder0904** [comment]
`brew install travis` would be better especially since its been 4 years now for such request. Thank you @the-vampiire for the solution. Also, a simple message on the site about how to install would be better. It was pain finding it (took me 5 mins 😂).

**Comment by ncray** [comment]
Looks like `brew install travis` is working now.

**Comment by waleCloud** [comment]
OMG 😮 after spending the whole trying to get gem install travis to work.

**Comment by gnzlbg** [comment]
> Looks like `brew install travis` is working now.

This is not documented anywhere. 

@BanzaiMan can somebody re-open this issue ?

**Comment by bhoudu** [comment]
I had troubles with installing travis through gem too (ffi notably brings troubles).

'brew install travis' seems the most reliable way to install the CLI so it would be better to spread the word for all.

It is truly the hidden gem.

**Comment by dgtlmonk** [comment]
The happiest path for OSX is to use `brew install travis`

**Comment by bitsofinfo** [comment]
annoying docs don't have this

**Comment by vjkoskela** [comment]
Seems like this should also update:

https://docs.travis-ci.com/user/encryption-keys/#usage
https://docs.travis-ci.com/user/encryption-keys/#usage

As well.
